### DISCUSSION TITLE
What difference between apache-airflow[mysql] and apache-airflow-providers-mysql

### DISCUSSION THREAD

**Discussion by HuFei2017** [apache/airflow#13525] [Q&A] [answered]
As the title says, I get troubles in the learning process.
I wanna know what difference between apache-airflow[mysql] and apache-airflow-providers-mysql, which decide what I should do with pip.

**Comment by turbaszek** [comment, marked as answer]
`apache-airflow-providers-mysql` is a stand-alone package which does not include "airflow core" it includes only content of `airflow/providers/mysql`. You can learn more about providers here:
http://airflow.apache.org/docs/apache-airflow-providers/

On the other hand `apache-airflow[mysql]` will install "core airflow" with additional dependencies required to run Airflow with mysql database as backend and - if I'm not mistaken - it will also install `apache-airflow-providers-mysql`. 

And if I correctly understand both commands should give similar result:
```
pip install apache-airflow[mysql]
# or
pip install apache-airflow
pip install apache-airflow-providers-mysql
```
@potiuk can you confirm?

**Reply by potiuk** [reply]
Yes. Those two are equivalent.

The updated documentation out extras and additional packaged (will be released in 2.0.1) can be found here. You can read more information there about extras/provider packages: https://github.com/apache/airflow/blob/master/docs/apache-airflow/extra-packages-ref.rst.
